crates/ruff_python_formatter/src/lib.rs

```diff
@@ -11,17 +15,85 @@ use crate::newlines::normalize_newlines;
 use crate::parentheses::normalize_parentheses;
 
 mod attachment;
+pub mod auxiliary;
 pub mod cli;
 pub mod context;
 mod cst;
 mod format;
+mod generated;
 mod newlines;
 mod parentheses;
 mod prelude;
 mod trivia;
 
 include!("../../ruff_formatter/shared_traits.rs");
 
+pub(crate) type PythonFormatter<'buf, 'ast> = Formatter<'buf, ASTFormatContext<'ast>>;
+
+/// Rule for formatting a JavaScript [`AstNode`].
+pub(crate) trait FormatNodeRule<N>
+where
+    N: AstNode,
+{
+    fn fmt(&self, node: &N, f: &mut PythonFormatter) -> FormatResult<()> {
+        if self.is_suppressed(node, f) {
+            todo!()
+        }
+
+        self.fmt_leading_comments(node, f)?;
+        self.fmt_node(node, f)?;
+        self.fmt_dangling_comments(node, f)?;
+        self.fmt_trailing_comments(node, f)
+    }
+
+    /// Formats the node without comments. Ignores any suppression comments.
+    fn fmt_node(&self, node: &N, f: &mut PythonFormatter) -> FormatResult<()> {
+        let needs_parentheses = self.needs_parentheses(node);
+
+        if needs_parentheses {
+            write!(f, [text("(")])?;
+        }
+
+        self.fmt_fields(node, f)?;
+
+        if needs_parentheses {
+            write!(f, [text(")")])?;
+        }
+
+        Ok(())
+    }
+
+    /// Formats the node's fields.
+    fn fmt_fields(&self, item: &N, f: &mut PythonFormatter) -> FormatResult<()>;
+
+    /// Formats the [leading comments](rome_formatter::comments#leading-comments) of the node.
+    ///
+    /// You may want to override this method if you want to manually handle the formatting of comments
+    /// inside of the `fmt_fields` method or customize the formatting of the leading comments.
+    fn fmt_leading_comments(&self, _node: &N, _f: &mut PythonFormatter) -> FormatResult<()> {
+        todo!()
```

Comment:
Nit: Maybe return `Ok(())` for now so that it simply drops all comments for now (instead of throwing).